test/unit/quantum/results/test_reckoning.py

```diff
@@ -32,40 +32,76 @@ class TestExpvalReckoner:
     """Test ExpvalReckoner interface."""
 
     @mark.parametrize(
-        "counts, expected",
+        "data, expected",
         [
-            ({}, (Counts({}),)),
-            (Counts({}), (Counts({}),)),
-            ({0: 1}, (Counts({0: 1}),)),
-            (Counts({0: 1}), (Counts({0: 1}),)),
-            ({0: 0, 1: 1}, (Counts({0: 0, 1: 1}),)),
-            (Counts({0: 0, 1: 1}), (Counts({0: 0, 1: 1}),)),
-            ([{}], (Counts({}),)),
-            ([Counts({})], (Counts({}),)),
-            ([{0: 1}], (Counts({0: 1}),)),
-            ([Counts({0: 1})], (Counts({0: 1}),)),
-            ([{0: 0, 1: 1}], (Counts({0: 0, 1: 1}),)),
-            ([Counts({0: 0, 1: 1})], (Counts({0: 0, 1: 1}),)),
-            ([{}, {0: 0, 1: 1}], (Counts({}), Counts({0: 0, 1: 1}))),
-            ([Counts({}), {0: 0, 1: 1}], (Counts({}), Counts({0: 0, 1: 1}))),
-            ([{}, Counts({0: 0, 1: 1})], (Counts({}), Counts({0: 0, 1: 1}))),
-            ([Counts({}), Counts({0: 0, 1: 1})], (Counts({}), Counts({0: 0, 1: 1}))),
+            ({}, (QuasiDistribution({}),)),
+            ({0: 1}, (QuasiDistribution({0: 1}),)),
+            ({0: 0, 1: 1}, (QuasiDistribution({0: 0, 1: 1}),)),
         ],
     )
-    def test_validate_counts_list(self, counts, expected):
-        """Test validate counts."""
-        valid = ExpvalReckoner._validate_counts_list(counts)
+    def test_validate_frequencies(self, data, expected):
+        """Test validate frequencies."""
+        for frequencies in (data, Counts(data), QuasiDistribution(data)):
+            valid = ExpvalReckoner._validate_frequencies_list(frequencies)
+            assert isinstance(valid, tuple)
+            assert all(isinstance(c, QuasiDistribution) for c in valid)
+            assert valid == expected
```

Comment:
You are testing `validate_frequencies` but calling `validate_frequencies_list` (which you test afterwards). Therefore `expected` should not be a `tuple`.

PS. That `for` loop idea is very smart 😉